server/server.go

```diff
@@ -103,8 +120,24 @@ func (s *RPCServer) AddMiddleware(m MiddlewareFunc) *RPCServer {
 }
 
 // AddHandler adds a new handler to the RPC server.
-func (s *RPCServer) AddHandler(queueName string, handler HandlerFunc) {
-	s.handlers[queueName] = handler
+func (s *RPCServer) AddHandler(routingKey string, handler HandlerFunc) {
+	s.AddExchangeHandler(routingKey, "", "direct", amqp.Table{}, handler)
+}
+
+// AddExchangeHandler will add a handler for any type of exchange and routing
+// key. If no values are given for routing key or exchange name the default
+// empty string will be used. Custom bind headers are passed as amqp.Table just
+// like they are treated when binding the queue.
+func (s *RPCServer) AddExchangeHandler(routingKey, exchangeName, exchangeType string, bindHeaders amqp.Table, handler HandlerFunc) {
```

Comment:
How about using a struct instad of all these arguments? That way we can pre-define some defaults:

```go
server.AddExchangeHandler(DirectBinding("my.routing.key"), handler)
server.AddExchangeHandler(FanoutBinding("my.routing.key"), handler)
server.AddExchangeHandler(TopicBinding("my.routing.#"), handler)
```

Reply:
Yeah why not!? But I guess we should use `FanoutBinding("my.exchange")` since the routing key isn't used for fanout queues. :)